mms_msg/sampling/pattern/meeting/state_based/transition_model.py

```diff
@@ -194,6 +248,45 @@ def __repr__(self):
             f"Probability matrix:\n {self.probability_matrix}"
         )
 
+    @staticmethod
+    def to_json(obj: MarkovModel) -> str:
+        def json_default(o):
+            if isinstance(o, np.ndarray):
+                return o.tolist()
+            else:
+                return o.__dict__
+        return json.dumps(obj, default=json_default)
+
+    @staticmethod
+    def from_json(json_string: str) -> MarkovModel:
+        obj = MarkovModel(np.ones((1, 1)))
+        data = json.loads(json_string)
+        for k, v in data.items():
+            if k == 'probability_matrix':
+                obj.__dict__[k] = np.asarray(v)
+            else:
+                obj.__dict__[k] = v
+        return obj
+
+    @staticmethod
+    def save(obj: MarkovModel, filepath: Optional[str] = 'state_transition_model.json') -> None:
+        with open(filepath, 'w+') as file:
+            try:
+                json_string = MarkovModel.to_json(obj)
+                file.write(json_string)
+            finally:
+                file.close()
```

Comment:
You don't need this here, closing the file is already done in the file context manager.